test/integration/offchain/match2.test.ts

```diff
@@ -316,37 +324,46 @@ describe('match2', () => {
     it('incorrect state when creating on-chain - 400', async () => {
       const response = await post(app, `/v1/match2/${seededMatch2AcceptedA}/proposal`, {})
       expect(response.status).to.equal(400)
-      expect(response.body).to.equal(`Match2 must have state: 'pending'`)
+      expect(response.body).to.equal(`state must be pending, is: acceptedA`)
     })
 
     it('demandA missing token ID - 400', async () => {
+      // const createMatch2 = await post(app, '/v1/match2', {
+      //   demandA: seededDemandAMissingTokenId,
+      //   demandB: seededDemandBId,
+      // })
+      // expect(createMatch2.status).to.equal(201) //the above will not be created
+
+      // const response = await post(app, `/v1/match2/${createMatch2.body.id}/proposal`, {})
+      // expect(response.status).to.equal(400)
+      // expect(response.body).to.equal('state must be created, is: pending')
+
+      //====================== UPDATED: =====================
       const createMatch2 = await post(app, '/v1/match2', {
         demandA: seededDemandAMissingTokenId,
-        demandB: seededDemandBId,
+        demandB: seededDemandBWithTokenId,
       })
-      expect(createMatch2.status).to.equal(201)
-
-      const response = await post(app, `/v1/match2/${createMatch2.body.id}/proposal`, {})
-      expect(response.status).to.equal(400)
-      expect(response.body).to.equal('DemandA must be on chain')
+      expect(createMatch2.status).to.equal(400)
```

Comment:
Is this now testing something different? The failure `400` has moved onto the other API

Reply:
So this now fails in the 1st step, when we are trying to propose a new match (this is because the seeded demandAMissingTokenId is in pending state and we are checking that it is 'created') - it does not get to line 337 like it did before